### PR TITLE
core-lightning 23.05 (new formula)

### PR DESCRIPTION
**Pull request by sugarjig**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This is [Core Lightning](https://github.com/ElementsProject/lightning), a Lightning Network implementation focusing on spec compliance and performance

### CODE REVIEW

File: Formula/core-lightning.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+class CoreLightning < Formula
+  desc "Lightning Network implementation focusing on spec compliance and performance"
+  homepage "https://github.com/ElementsProject/lightning"
+  url "https://github.com/ElementsProject/lightning.git",
+      using:    :git,
```
Comment (carlocab) [inline Formula/core-lightning.rb:5]:
```suggestion
```
This isn't needed.

File: Formula/core-lightning.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+class CoreLightning < Formula
+  desc "Lightning Network implementation focusing on spec compliance and performance"
+  homepage "https://github.com/ElementsProject/lightning"
+  url "https://github.com/ElementsProject/lightning.git",
```
Comment (carlocab) [inline Formula/core-lightning.rb:4]: Do we need a Git clone? Can we use a tarball?

File: Formula/core-lightning.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  depends_on "gettext" => :build
+  depends_on "gmp" => :build
+  depends_on "gnu-sed" => :build
+  depends_on "libsodium" => :build
+  depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
+  depends_on "poetry" => :build
+  depends_on "postgresql@14" => :build
+  depends_on "protobuf" => :build
+  depends_on "python@3.8" => :build
```
Comment (carlocab) [inline Formula/core-lightning.rb:21]: Can we use a newer version of Python?

File: Formula/core-lightning.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  depends_on "bitcoin"
+
+  def install
+    File.open("external/lowdown/configure.local", "a") do |configure_local|
+      configure_local.puts "HAVE_SANDBOX_INIT=0"
+    end
+    system "poetry", "env", "use", "3.8"
+    system "poetry", "install"
```
Comment (carlocab) [inline Formula/core-lightning.rb:29]: Where does this install the required Python libraries?

**Comment by carlocab** [comment]
Could you squash your commits together as well? Thanks!

File: Formula/core-lightning.rb (hunk 2 above)
Comment (sugarjig) [reply Formula/core-lightning.rb:4]: The Git repo contains submodules. The build fails if they are not present.

File: Formula/core-lightning.rb (hunk 3 above)
Comment (sugarjig) [reply Formula/core-lightning.rb:21]: The build depends on this specific version. I don't fully understand why.

File: Formula/core-lightning.rb (hunk 4 above)
Comment (sugarjig) [reply Formula/core-lightning.rb:29]: Poetry manages a virtualenv directory that contains the libraries.

File: Formula/core-lightning.rb (hunk 2 above)
Comment (carlocab) [reply Formula/core-lightning.rb:4]: What are in the submodules?

File: Formula/core-lightning.rb (hunk 4 above)
Comment (carlocab) [reply Formula/core-lightning.rb:29]: Where does it put the virtualenv directory?

Comment (sugarjig) [reply Formula/core-lightning.rb:29]: `~/Library/Caches/pypoetry/virtualenvs`

Comment (carlocab) [reply Formula/core-lightning.rb:29]: Are these libraries needed at run-time?

File: Formula/core-lightning.rb (hunk 2 above)
Comment (sugarjig) [reply Formula/core-lightning.rb:4]: Dependencies
- gheap
- jsmn
- libbacktrace
- libsodium
- libwally-core
- lnprototest
- lowdown

File: Formula/core-lightning.rb (hunk 4 above)
Comment (sugarjig) [reply Formula/core-lightning.rb:28]: Not to my knowledge. This is a C app.

File: Formula/core-lightning.rb (hunk 2 above)
Comment (carlocab) [reply Formula/core-lightning.rb:4]: We should use Homebrew-provided dependencies where available (e.g. libsodium) instead of what's in the submodules.

Comment (sugarjig) [reply Formula/core-lightning.rb:4]: We are in the case of libsodium. The others are not in Homebrew.

File: Formula/core-lightning.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "gettext" => :build
+  depends_on "gmp" => :build
+  depends_on "gnu-sed" => :build
+  depends_on "libsodium" => :build
+  depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
+  depends_on "poetry" => :build
+  depends_on "postgresql@14" => :build
```
[the remark this answers is not in the document]
Comment (sugarjig) [reply Formula/core-lightning.rb]: Replaced with `uses_from_macos "sqlite"`. It can't be built with either.

**Comment by sugarjig** [comment]
@carlocab The checks are all passing now. Are there any more outstanding questions?

**Comment by sugarjig** [comment]
@carlocab I tried your suggestion of removing the libsodium submodule, but the Makefile is written to restore the submodules if any are missing, which overwrites the change we have to make to the lowdown submodule. Is it okay to move forward with the formula as-is? As a follow up, I will look into creating formulae for all the dependencies, hopefully removing the need for submodules.

**Review by carlocab — APPROVED** [review]
That works for me. Thanks!

**Comment by sugarjig** [comment]
@carlocab When will this be merged?

**Pull request merged**